recipe/meta.yaml

```diff
@@ -48,19 +51,19 @@ outputs:
         - libevent
         - libnl  # [linux]
         - zlib
-        - ucx  # [enable_ucx]
-        - cuda-version {{ cuda_version }}  # [enable_cuda]
+        - ucx  # [linux]
+        - cuda-version {{ cuda_compiler_version }}  # [cuda_compiler != "None"]
       run:
         - mpi 1.0 openmpi
         #- openpmix
         #- prrte
       run_constrained:
-        - {{ pin_compatible("ucx", max_pin="x.x") }}  # [enable_ucx]
+        - {{ pin_compatible("ucx", max_pin="x.x") }}  # [linux]
         # Open MPI only uses CUDA Driver APIs, set the minimal driver version
-        - __cuda  >= {{ cuda_version }}  # [enable_cuda]
+        - __cuda  >= {{ cuda_major ~ ".0" }}  # [cuda_compiler != "None"]
         # Ensure a consistent CUDA environment
-        - cuda-version  >= {{ cuda_version }}  # [enable_cuda]
-        - cudatoolkit   >= {{ cuda_version }}  # [enable_cuda]
+        - cuda-version  >= {{ cuda_major ~ ".0" }}  # [cuda_compiler != "None"]
+        - cudatoolkit   >= {{ cuda_major ~ ".0" }}  # [cuda_compiler != "None"]
```

Comment:
I think we made a mistake and this can be just removed since we rely only on the driver; if UCX needs anything we'll get it transitively too.
```suggestion
```

Reply:
Yeah guess it depends on whether we want the CUDA libraries that a user chooses to installs to match what the OpenMPI package was built to support (as this is only a soft dependency)

Maybe two cases for comparison would be Arrow & Numba:

* https://github.com/conda-forge/arrow-cpp-feedstock/pull/1079
* https://github.com/conda-forge/numba-feedstock/pull/115/commits/4adee2186ec1d5aafc582d79ee8853145aaba8a0

Ok with whatever you and Lisandro think is best 🙂

Reply:
We already use `cuda-version`, so this wouldn't be needed. Also, this doesn't work for CUDA 12 anyway. Let's just remove it.